Some details are already shown in #18074, but I append some more logs so you can see where it happens. This is one of the logs where we tried to run `make deb` on a Scaleway C1 machine that was created with the Docker_1_9_0 image.

The dynamic build takes some minutes at the message "Building: bundles/1.10.0-dev/dynbinary/docker-1.10.0-dev" and then the error occurs in the middle of compiling/linking.

```
...
Step 34 : ENV RSRC_COMMIT e48dbf1b7fc464a9e85fcec450dddf80816b76e0
 ---> Using cache
 ---> a40a311d571a
Step 35 : RUN set -x     && git clone https://github.com/akavel/rsrc.git /go/src/github.com/akavel/rsrc     && cd /go/src/github.com/akavel/rsrc     && git checkout -q $RSRC_COMMIT     && go install -v
 ---> Using cache
 ---> d54d1d0a1434
Step 36 : ENTRYPOINT hack/dind
 ---> Using cache
 ---> 33630ed55542
Step 37 : COPY . /go/src/github.com/docker/docker
 ---> Using cache
 ---> f42762da91eb
Successfully built f42762da91eb
docker run --rm -i --privileged -e BUILDFLAGS -e DOCKER_CLIENTONLY -e DOCKER_DEBUG -e DOCKER_EXPERIMENTAL -e DOCKER_REMAP_ROOT -e DOCKER_GRAPHDRIVER -e DOCKER_STORAGE_OPTS -e DOCKER_USERLANDPROXY -e DOCKER_ENGINE_OSARCH -e DOCKER_ENGINE_GOOS -e DOCKER_ENGINE_GOARCH -e DOCKER_FILE -e TESTDIRS -e TESTFLAGS -e TIMEOUT -v "/var/lib/jenkins/workspace/pr3 docker-dev (build-dyndeb)/bundles:/go/src/github.com/docker/docker/bundles" "docker-dev:HEAD" hack/make.sh dynbinary build-deb

bundles/1.10.0-dev already exists. Removing.

---> Making bundle: dynbinary (in bundles/1.10.0-dev/dynbinary)
Created binary: bundles/1.10.0-dev/dynbinary/dockerinit-1.10.0-dev
Building: bundles/1.10.0-dev/dynbinary/docker-1.10.0-dev
# github.com/docker/docker/docker
/var/tmp/go-link-Zgk7oN/000008.o: In function `__gnu_uldivmod_helper':
:(.text+0x87b20): multiple definition of `__gnu_uldivmod_helper'
/var/tmp/go-link-Zgk7oN/000003.o::(.text+0x638): first defined here
/var/tmp/go-link-Zgk7oN/000008.o: In function `__divdi3':
:(.text+0x87b50): multiple definition of `__divdi3'
/var/tmp/go-link-Zgk7oN/000003.o::(.text+0x668): first defined here
/var/tmp/go-link-Zgk7oN/000008.o: In function `__aeabi_uldivmod':
:(.text+0x87a40): multiple definition of `__aeabi_uldivmod'
/var/tmp/go-link-Zgk7oN/000003.o::(.text+0x5cc): first defined here
/var/tmp/go-link-Zgk7oN/000008.o: In function `__aeabi_uidiv':
:(.text+0x87170): multiple definition of `__udivsi3'
/var/tmp/go-link-Zgk7oN/000003.o::(.text+0xcc0): first defined here
/var/tmp/go-link-Zgk7oN/000008.o: In function `__udivdi3':
:(.text+0x87eac): multiple definition of `__udivdi3'
/var/tmp/go-link-Zgk7oN/000003.o::(.text+0x9c4): first defined here
/var/tmp/go-link-Zgk7oN/000008.o: In function `__aeabi_uidiv':
:(.text+0x87170): multiple definition of `__aeabi_uidiv'
/var/tmp/go-link-Zgk7oN/000003.o::(.text+0xcc0): first defined here
/var/tmp/go-link-Zgk7oN/000008.o: In function `__aeabi_uidivmod':
:(.text+0x873cc): multiple definition of `__aeabi_uidivmod'
/var/tmp/go-link-Zgk7oN/000003.o::(.text+0xf1c): first defined here
/var/tmp/go-link-Zgk7oN/000008.o: In function `__gnu_ldivmod_helper':
:(.text+0x87af0): multiple definition of `__gnu_ldivmod_helper'
/var/tmp/go-link-Zgk7oN/000003.o::(.text+0x608): first defined here
collect2: error: ld returned 1 exit status
/usr/local/go/pkg/tool/linux_arm/5l: running gcc failed: unsuccessful exit status 0x100
Makefile:77: recipe for target 'deb' failed
make: *** [deb] Error 1
Build step 'Execute shell' marked build as failure
Archiving artifacts
Finished: FAILURE
```

--- a/daemon/logger/journald/journald.go
+++ b/daemon/logger/journald/journald.go
@@ -1,4 +1,4 @@
-// +build linux
+// +build linux,!arm
 
 // Package journald provides the log driver for forwarding server logs
 // to endpoints that receive the systemd format.

--- a/daemon/logger/journald/journald_unsupported.go
+++ b/daemon/logger/journald/journald_unsupported.go
@@ -1,4 +1,4 @@
-// +build !linux
+// +build !linux linux,arm
 
 package journald
 

--- a/daemon/logger/journald/read.go
+++ b/daemon/logger/journald/read.go
@@ -1,4 +1,4 @@
-// +build linux,cgo,!static_build,journald
+// +build linux,cgo,!static_build,journald,!arm
 
 package journald
 

--- a/daemon/logger/journald/read_unsupported.go
+++ b/daemon/logger/journald/read_unsupported.go
@@ -1,4 +1,4 @@
-// +build !linux !cgo static_build !journald
+// +build !linux !cgo static_build !journald linux,arm
 
 package journald
 

